helix-core/src/test/java/org/apache/helix/task/TestGetLastScheduledTaskExecInfo.java

```diff
@@ -41,37 +42,47 @@ public void beforeClass() throws Exception {
   }
 
   @Test
-  public void testGetLastScheduledTaskExecInfo() throws InterruptedException {
-    List<Long> startTimesWithStuckTasks = setupTasks("TestWorkflow_2", 5, 99999999);
+  public void testGetLastScheduledTaskExecInfo() throws Exception {
+    // Start new queue that has one job with long tasks and record start time of the tasks
+    String queueName = TestHelper.getTestMethodName();
+    List<Long> startTimesWithStuckTasks = setupTasks(queueName, 5, 99999999, 2);
     // Wait till the job is in progress
-    _driver.pollForJobState("TestWorkflow_2", "TestWorkflow_2_job_0", TaskState.IN_PROGRESS);
+    _driver.pollForJobState(queueName, queueName + "_job_0", TaskState.IN_PROGRESS);
 
     // First two must be -1 (two tasks are stuck), and API call must return the last value (most
     // recent timestamp)
     Assert.assertEquals(startTimesWithStuckTasks.get(0).longValue(), INVALID_TIMESTAMP);
     Assert.assertEquals(startTimesWithStuckTasks.get(1).longValue(), INVALID_TIMESTAMP);
-    Long lastScheduledTaskTs = _driver.getLastScheduledTaskTimestamp("TestWorkflow_2");
-    // Get the last timestamp (which will be most recent)
-    Assert.assertEquals(startTimesWithStuckTasks.get(4), lastScheduledTaskTs);
 
-    TaskExecutionInfo execInfo = _driver.getLastScheduledTaskExecutionInfo("TestWorkflow_2");
-    Assert.assertEquals(execInfo.getJobName(), "TestWorkflow_2_job_0");
-    // Workflow 2 will be stuck, so its partition state will be RUNNING
-    Assert.assertEquals(execInfo.getTaskPartitionState(), TaskPartitionState.RUNNING);
-    Assert.assertEquals(execInfo.getStartTimeStamp(), lastScheduledTaskTs);
+    // Workflow will be stuck so its partition state will be Running
+    boolean hasQueueReachedDesiredState = TestHelper.verify(() -> {
+      Long lastScheduledTaskTs = _driver.getLastScheduledTaskTimestamp(queueName);
+      TaskExecutionInfo execInfo = _driver.getLastScheduledTaskExecutionInfo(queueName);
+      return (execInfo.getJobName().equals(queueName + "_job_0")
+          && execInfo.getTaskPartitionState() == TaskPartitionState.RUNNING
+          && execInfo.getStartTimeStamp().equals(lastScheduledTaskTs)
+          && startTimesWithStuckTasks.get(4).equals(lastScheduledTaskTs));
+    }, 30 * 1000);
+    Assert.assertTrue(hasQueueReachedDesiredState);
 
-    List<Long> startTimesFastTasks = setupTasks("TestWorkflow_3", 4, 10);
-    // Wait till the job is in progress
-    _driver.pollForJobState("TestWorkflow_3", "TestWorkflow_3_job_0", TaskState.IN_PROGRESS);
-    // API call needs to return the most recent timestamp (value at last index)
-    lastScheduledTaskTs = _driver.getLastScheduledTaskTimestamp("TestWorkflow_3");
-    execInfo = _driver.getLastScheduledTaskExecutionInfo("TestWorkflow_3");
+    // Stop and delete the queue
+    _driver.stop(queueName);
+    _driver.deleteAndWaitForCompletion(queueName, 30 * 1000);
```

Comment:
Do you think you could make recurring 30 * 1000's a constant?

Reply:
Fixed.